presto-main/src/main/java/com/facebook/presto/operator/OperatorContext.java

```diff
@@ -634,6 +635,13 @@ private long decrementSpilledReservation(long reservedBytes, long bytesBeingFree
         }
 
         @Override
+        public void close()
+        {
+            // Only products of SpillContext.newLocalSpillContext() should be closed.
+            throw new UnsupportedOperationException(format("%s should be closed directly", getClass()));
```

Comment:
I do not get the meaning of the message.

Also this could be moved to #8263 and merged sooner.

Also maybe we should get rid of default implementation of close() in `SpillContext` interface.
If you decide to do so also add it to #8263